src/types.ts

```diff
@@ -81,289 +81,328 @@ export type ProximityRule = {
+   * Custom message for when no results are found.
    *
-   * Default value is `undefined` - use all countries.
+   * Default:
+   * ```
+   * "Oops! Looks like you're trying to predict something that's not quite right.
+   * We can't seem to find what you're looking for. Maybe try double-checking your spelling or try a different search term.
+   * Keep on typing - we'll do our best to get you where you need to go!"
+   * ```
    */
-  country?: string | string[];
+  noResultsMessage?: string;
 
   /**
-   * Filter of feature types to return.
+   * Style of the picked result on the map:
+   * - `"marker-only"`: Show only a marker at the center of the feature.
+   * - `"full-geometry"`: Display the full feature geometry.
+   * - `"full-geometry-including-polygon-center-marker"`: Display full geometry with a marker at the polygon center.
    *
-   * Default value is `undefined` - all available feature types are returned.
+   * Default: `"full-geometry"`.
    */
-  types?: string[];
+  pickedResultStyle?: PickedResultStyle;
 
   /**
-   * Use `limit` value for reverse geocoding even if `types` is not an array with a single element.
-   * Will work only if enabled on the server.
+   * Custom placeholder for the input box.
    *
-   * Default value is `false`.
+   * Default: `"Search"`.
    */
-  exhaustiveReverseGeocoding?: boolean;
+  placeholder?: string;
 
   /**
-   * If set to `true` then use all types except for those listed in `types`.
+   * Prioritizes search results closer to a proximity point.
+   * The first matching rule in the array is used.
+   * Set to `null` to disable proximity.
    *
-   * Default value is `false`.
+   * Default: `[ { type: "server-geolocation" } ]`.
    */
-  excludeTypes?: boolean;
+  proximity?: ProximityRule[] | null;
 
   /**
-   * Geocoding API URL.
+   * Activates reverse geocoding.
    *
-   * Default value is MapTiler Geocoding API URL.
+   * Default: `false`.
    */
-  apiUrl?: string;
+  reverseActive?: boolean;
 
   /**
-   * Extra fetch parameters.
+   * Title of the reverse geocoding toggle button.
    *
-   * Default value is `undefined`.
+   * Default: `"toggle reverse geocoding"`.
    */
-  fetchParameters?: RequestInit;
+  reverseButtonTitle?: string;
 
   /**
-   * Base URL for POI icons.
+   * Excludes types for reverse geocoding.
    *
-   * Default value is `"icons/"` for Svelte apps, otherwise `"https://cdn.maptiler.com/maptiler-geocoding-control/v${version}/icons/"`.
+   * Default: Same as `excludeTypes`.
    */
-  iconsBaseUrl?: string;
+  reverseGeocodingExcludeTypes?: boolean;
 
   /**
-   * Function to adjust URL search parameters.
+   * Limits results for reverse geocoding.
+   * Applied only if effective types contain a single value.
    *
-   * Default value is empty function.
+   * See also `reverseGeocodingTypes` option.
    *
-   * @deprecated use `adjustUrl`
+   * Default: Value of the `limit` option if set, or `1` for a single effective type.
```

Comment:
@zdila Thanks a lot for your changes. I had a bit of trouble understanding this. Could it be:
 Default: Value of the `limit` option if set, or `1` for each effective type otherwise.

Reply:
Fair enough, then it should be 
 Default: Value of the `limit` option if set, or `1` otherwise
Since you already mention above 
> Applied only if effective types contain a single value.

Reply:
Updated.